producer-c/producer-cloudwatch-integ/canary/CanaryStreamUtils.cpp

```diff
@@ -191,31 +192,49 @@ STATUS publishErrorRate(STREAM_HANDLE streamHandle, PCanaryStreamCallbacks pCana
     STATUS retStatus = STATUS_SUCCESS;
     StreamMetrics canaryStreamMetrics;
     canaryStreamMetrics.version = STREAM_METRICS_CURRENT_VERSION;
-    Aws::CloudWatch::Model::MetricDatum putFrameErrorRateDatum, aggputFrameErrorRateDatum, errorAckDatum, aggErrorAckDatum;
+    Aws::CloudWatch::Model::MetricDatum putFrameErrorRateDatum, aggputFrameErrorRateDatum, errorAckDatum, aggErrorAckDatum, totalErrorDatum, aggTotalErrorDatum;
+    DOUBLE putFrameErrorRate, errorAckRate;
+    UINT64 numberOfPutFrameErrors, numberOfErrorAcks;
     CHK(pCanaryStreamCallbacks != NULL, STATUS_NULL_ARG);
     CHK_STATUS(getKinesisVideoStreamMetrics(streamHandle, &canaryStreamMetrics));
-    DOUBLE putFrameErrorRate, errorAckRate;
-    putFrameErrorRate = (DOUBLE) (canaryStreamMetrics.putFrameErrors - pCanaryStreamCallbacks->historicStreamMetric.prevPutFrameErrorCount) / (DOUBLE) duration;
+
+    numberOfPutFrameErrors = canaryStreamMetrics.putFrameErrors - pCanaryStreamCallbacks->historicStreamMetric.prevPutFrameErrorCount;
+    numberOfErrorAcks = canaryStreamMetrics.errorAcks - pCanaryStreamCallbacks->historicStreamMetric.prevErrorAckCount;
+    putFrameErrorRate = (DOUBLE) (numberOfPutFrameErrors) / (DOUBLE) (duration / HUNDREDS_OF_NANOS_IN_A_SECOND);
```

Comment:
Make sure duration is not 0

Reply:
Duration cannot be zero because this function does not get called until duration is greater than 60 seconds.